wagtail/images/views/multiple.py

```diff
@@ -43,6 +47,33 @@ def get_upload_form_class(self):
     def get_edit_form_class(self):
         return get_image_multi_form(self.model)
 
+    def get_confirm_duplicate_upload_response(self, duplicates):
+        return render_to_string(
+            "wagtailimages/images/confirm_duplicate_upload.html",
+            {
+                "existing_image": duplicates[0],
+                "delete_action": reverse(
+                    self.delete_object_url_name, args=(self.object.id,)
+                ),
+            },
+            request=self.request,
+        )
+
+    def get_edit_object_response_data(self):
+        data = super().get_edit_object_response_data()
+        duplicates = find_image_duplicates(self.model, self.object)
```

Comment:
We don't account for the fact that the current user may not have `choose` permission for those duplicates - I think a duplicate is probably only relevant if they're actually allowed to choose it on a page

Reply:
I have updated the `find_image_duplicates` function to take this into account. Thanks!